src/main/java/com/thebois/models/beings/AbstractBeing.java

```diff
@@ -1,59 +1,37 @@
 package com.thebois.models.beings;
 
-import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Objects;
-import java.util.Optional;
-import java.util.Stack;
 
-import com.google.common.eventbus.Subscribe;
-
-import com.thebois.Pawntastic;
-import com.thebois.listeners.events.ObstaclePlacedEvent;
 import com.thebois.models.Position;
-import com.thebois.models.beings.pathfinding.IPathFinder;
 import com.thebois.models.beings.roles.AbstractRole;
-import com.thebois.models.beings.roles.RoleFactory;
+import com.thebois.models.inventory.IInventory;
+import com.thebois.models.inventory.items.IItem;
 
 /**
- * An abstract implementation of IBeing.
+ * An independent agent that can act in the world according to its assigned role.
  */
-public abstract class AbstractBeing implements IBeing {
+public abstract class AbstractBeing implements IBeing, IActionPerformer {
 
     // The max speed of the AbstractBeing
     private static final float MAX_WALKING_DISTANCE = 0.1f;
-    private final IPathFinder pathFinder;
-    private Stack<Position> path;
+    private final IInventory inventory;
     private Position position;
     private AbstractRole role;
+    private Position destination;
```

Comment:
Question

Guessing we use this to see if we have reached our destination, would make sense. But do we REALLY need it here?

Reply:
It's not *the* actual final destination, it's the next tile that the being is moving towards. This step-towards-the-final-destination is the being's temporary destination, given to it from the `MoveAction` which holds the *actual* final destination.

Reply:
👍